<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Lei Sun" />

<meta name="date" content="2018-12-12" />

<title>Empirical CDF of simulated $z$-scores</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">cashr paper</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/LSun/cashr_paper">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Empirical CDF of simulated <span class="math inline">\(z\)</span>-scores</h1>
<h4 class="author"><em>Lei Sun</em></h4>
<h4 class="date"><em>2018-12-12</em></h4>

</div>


<p><strong>Last updated:</strong> 2018-12-14</p>
<strong>workflowr checks:</strong> <small>(Click a bullet for more information)</small>
<ul>
<li>
<details>
<p><summary> <strong style="color:blue;">✔</strong> <strong>R Markdown file:</strong> up-to-date </summary></p>
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</details>
</li>
<li>
<details>
<p><summary> <strong style="color:blue;">✔</strong> <strong>Environment:</strong> empty </summary></p>
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</details>
</li>
<li>
<details>
<p><summary> <strong style="color:blue;">✔</strong> <strong>Seed:</strong> <code>set.seed(777)</code> </summary></p>
<p>The command <code>set.seed(777)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</details>
</li>
<li>
<details>
<p><summary> <strong style="color:blue;">✔</strong> <strong>Session information:</strong> recorded </summary></p>
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</details>
</li>
<li>
<details>
<p><summary> <strong style="color:blue;">✔</strong> <strong>Repository version:</strong> <a href="https://github.com/LSun/cashr_paper/tree/97c0343dc757af97ba63c896287403304b73159d" target="_blank">97c0343</a> </summary></p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/FDP_sim_cache/
    Ignored:    analysis/cashr_paper_fig_deconv_cache/
    Ignored:    analysis/cashr_paper_fig_ecdf_cache/

Untracked files:
    Untracked:  analysis/FDP_sim.rmd

</code></pre>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</details>
</li>
</ul>
<details>
<summary> <small><strong>Expand here to see past versions:</strong></small> </summary>
<ul>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
File
</th>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
<th style="text-align:left;">
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/LSun/cashr_paper/blob/97c0343dc757af97ba63c896287403304b73159d/analysis/cashr_paper_fig_ecdf.rmd" target="_blank">97c0343</a>
</td>
<td style="text-align:left;">
LSun
</td>
<td style="text-align:left;">
2018-12-14
</td>
<td style="text-align:left;">
wflow_publish(all = TRUE)
</td>
</tr>
<tr>
<td style="text-align:left;">
rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/LSun/cashr_paper/blob/63cd00ace4d251f8b7b031d2a0e2eab029a82cf1/analysis/cashr_paper_fig_ecdf.rmd" target="_blank">63cd00a</a>
</td>
<td style="text-align:left;">
LSun
</td>
<td style="text-align:left;">
2018-12-14
</td>
<td style="text-align:left;">
figures
</td>
</tr>
<tr>
<td style="text-align:left;">
rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/LSun/cashr_paper/blob/c2f61205dd6cbbbdd58c584cfefa4c5b18cbc1a6/analysis/cashr_paper_fig_ecdf.rmd" target="_blank">c2f6120</a>
</td>
<td style="text-align:left;">
Lei Sun
</td>
<td style="text-align:left;">
2018-12-13
</td>
<td style="text-align:left;">
data
</td>
</tr>
</tbody>
</table>
</ul>
</details>
<hr />
<pre class="r"><code>## load the data
r &lt;- readRDS(&quot;../data/liver.rds&quot;)</code></pre>
<pre class="r"><code>## load the script for the pipeline
## to convert RNA-seq count data to summary statistics for each gene
## in differential expression studies
source(&quot;../code/RNAseq_pipeline.R&quot;)</code></pre>
<pre class="r"><code>## common simulation settings
ngene &lt;- 1e4
nsamp &lt;- 5
nsim &lt;- 1e4
group &lt;- rep(0 : 1, each = nsamp)

## N(0,1) CDF plot positions
x.plot &lt;- seq(-6, 6, by = 0.01)
y.plot.norm &lt;- pnorm(x.plot)

## DKW bounds for the empirical CDF of 1e4 iid N(0,1) samples
## with alpha = 1e-4
upper.norm &lt;- y.plot.norm + sqrt(log(2 / (1 / nsim)) / (2 * ngene))
lower.norm &lt;- y.plot.norm - sqrt(log(2 / (1 / nsim)) / (2 * ngene))

## choose the 1e4 most expressed genes
Y = lcpm(r)
subset = top_genes_index(ngene, Y)
r = r[subset,]</code></pre>
<div id="keep-gene-gene-correlations" class="section level2">
<h2>Keep gene-gene correlations</h2>
<pre class="r"><code>## simulate null z-scores from RNA-seq data
## KEEP correlation
set.seed(7)
z.list &lt;- list()
for (i in seq(nsim)) {
  ## generate data
  counts &lt;- r[, sample(ncol(r), 2 * nsamp)]
  summary &lt;- count_to_summary(counts, group)
  z.list[[i]] &lt;- summary$z
}</code></pre>
<pre class="r"><code>z.keep.cor &lt;- z.list
saveRDS(z.list, &quot;../output/z_keep_cor.rds&quot;)</code></pre>
<pre class="r"><code>z.keep.cor &lt;- readRDS(&#39;../output/z_keep_cor.rds&#39;)</code></pre>
<pre class="r"><code>nsim &lt;- length(z.keep.cor)

# setEPS()
# postscript(&quot;../figures/ecdf_keep_cor.eps&quot;, width = 5, height = 5)
png(&quot;../figures/ecdf_keep_cor.png&quot;, width = 5, height = 5, units = &quot;in&quot;, res = 1000)

par(mar = c(4.5, 4.5, 2.5, 1))

plot(0, type = &quot;n&quot;, xlim = c(-5, 5), ylim = c(0, 1), ylab = &quot;(Empirical) CDF&quot;, xlab = &quot;z-score&quot;, cex.lab = 2)
title(expression(&#39;(a): Correlation Kept&#39;), cex.main = 2)

for (i in seq(nsim)) {
  lines(ecdf(z.keep.cor[[i]]), lwd = 1, col = &quot;grey75&quot;)
}

lines(x.plot, y.plot.norm, lwd = 2, col = &quot;blue&quot;)

lines(x.plot[upper.norm &lt;= 1 &amp; upper.norm &gt;= 0], upper.norm[upper.norm &lt;= 1 &amp; upper.norm &gt;= 0], lty = 3, col = &quot;blue&quot;)
lines(x.plot[lower.norm &lt;= 1 &amp; lower.norm &gt;= 0], lower.norm[lower.norm &lt;= 1 &amp; lower.norm &gt;= 0], lty = 3, col = &quot;blue&quot;)

legend(&quot;bottomright&quot;, lwd = 2, col = c(&quot;grey75&quot;, &quot;blue&quot;), c(expression(&quot;F&quot;[p]), expression(&#39;N(0,1)&#39;)), bty = &quot;n&quot;, cex = 1.75, x.intersp = 0.5)</code></pre>
</div>
<div id="remove-gene-gene-correlations" class="section level2">
<h2>Remove gene-gene correlations</h2>
<pre class="r"><code>## simulate null z-scores from RNA-seq data
## REMOVE correlation
set.seed(7)
z.list &lt;- list()
for (i in seq(nsim)) {
  ## generate data
  counts &lt;- t(apply(r, 1, sample, 2 * nsamp))
  summary &lt;- count_to_summary(counts, group)
  z.list[[i]] &lt;- summary$z
}</code></pre>
<pre class="r"><code>z.remove.cor &lt;- z.list
saveRDS(z.list, &quot;../output/z_remove_cor.rds&quot;)</code></pre>
<pre class="r"><code>z.remove.cor &lt;- readRDS(&#39;../output/z_remove_cor.rds&#39;)</code></pre>
<pre class="r"><code>nsim &lt;- length(z.remove.cor)

# setEPS()
# postscript(&quot;../figures/ecdf_remove_cor.eps&quot;, width = 5, height = 5)
png(&quot;../figures/ecdf_remove_cor.png&quot;, width = 5, height = 5, units = &quot;in&quot;, res = 1000)

par(mar = c(4.5, 4.5, 2.5, 1))

plot(0, type = &quot;n&quot;, xlim = c(-5, 5), ylim = c(0, 1), ylab = &quot;(Empirical) CDF&quot;, xlab = &quot;z-score&quot;, cex.lab = 2)
title(expression(&#39;(b): Correlation Removed&#39;), cex.main = 2)

for (i in seq(nsim)) {
  lines(ecdf(z.remove.cor[[i]]), lwd = 1, col = &quot;grey75&quot;)
}

lines(x.plot, y.plot.norm, lwd = 2, col = &quot;blue&quot;)

lines(x.plot[upper.norm &lt;= 1 &amp; upper.norm &gt;= 0], upper.norm[upper.norm &lt;= 1 &amp; upper.norm &gt;= 0], lty = 3, col = &quot;blue&quot;)
lines(x.plot[lower.norm &lt;= 1 &amp; lower.norm &gt;= 0], lower.norm[lower.norm &lt;= 1 &amp; lower.norm &gt;= 0], lty = 3, col = &quot;blue&quot;)

legend(&quot;bottomright&quot;, lwd = 2, col = c(&quot;grey75&quot;, &quot;blue&quot;), c(expression(&quot;F&quot;[p]), expression(&#39;N(0,1)&#39;)), bty = &quot;n&quot;, cex = 1.75, x.intersp = 0.5)</code></pre>
</div>
<div id="independent-n01-z-scores" class="section level2">
<h2>Independent <span class="math inline">\(N(0,1)\)</span> <span class="math inline">\(z\)</span>-scores</h2>
<pre class="r"><code># setEPS()
# postscript(&quot;../figures/ecdf_iid.eps&quot;, width = 5, height = 5)
png(&quot;../figures/ecdf_iid.png&quot;, width = 5, height = 5, units = &quot;in&quot;, res = 1000)

par(mar = c(4.5, 4.5, 2.5, 1))

plot(0, type = &quot;n&quot;, xlim = c(-5, 5), ylim = c(0, 1), ylab = &quot;(Empirical) CDF&quot;, xlab = &quot;z-score&quot;, cex.lab = 2)
title(expression(&#39;(c): iid N(0,1) z-scores&#39;), cex.main = 2)

set.seed(777)
for (i in seq(nsim)) {
  lines(ecdf(rnorm(ngene)), lwd = 1, col = &quot;grey75&quot;)
}

lines(x.plot, y.plot.norm, lwd = 2, col = &quot;blue&quot;)

lines(x.plot[upper.norm &lt;= 1 &amp; upper.norm &gt;= 0], upper.norm[upper.norm &lt;= 1 &amp; upper.norm &gt;= 0], lty = 3, col = &quot;blue&quot;)
lines(x.plot[lower.norm &lt;= 1 &amp; lower.norm &gt;= 0], lower.norm[lower.norm &lt;= 1 &amp; lower.norm &gt;= 0], lty = 3, col = &quot;blue&quot;)

legend(&quot;bottomright&quot;, lwd = 2, col = c(&quot;grey75&quot;, &quot;blue&quot;), c(expression(&quot;F&quot;[p]), expression(&#39;N(0,1)&#39;)), bty = &quot;n&quot;, cex = 1.75, x.intersp = 0.5)</code></pre>
</div>
<div id="average-of-cdf-with-gene-gene-correlation-kept" class="section level2">
<h2>Average of CDF with gene-gene correlation kept</h2>
<pre class="r"><code>## calculate empirical CDF for every set of 1e4 z-scores at specified positions
## these positions are dense in the tails and loose in the center
quantile.left &lt;- exp(seq(-21, -5, by = 0.01))
quantile.center &lt;- seq(0.007, 0.993, by = 0.001)
quantile.right &lt;- rev(quantile.left)
emp.cdf.left &lt;- sapply(quantile.left, function(x) {sapply(z.keep.cor, function(y) mean(y &lt;= qnorm(x)))})
emp.cdf.center &lt;- sapply(quantile.center, function(x) {sapply(z.keep.cor, function(y) mean(y &lt;= qnorm(x)))})
emp.cdf.right &lt;- sapply(quantile.right, function(x) {sapply(z.keep.cor, function(y) mean(y &lt;= -qnorm(x)))})
emp.cdf.right.exceed &lt;- sapply(quantile.right, function(x) {sapply(z.keep.cor, function(y) mean(y &gt; -qnorm(x)))})</code></pre>
<pre class="r"><code>ecdf.avg.left &lt;- colMeans(emp.cdf.left)
ecdf.avg.center &lt;- colMeans(emp.cdf.center)
ecdf.avg.right &lt;- colMeans(emp.cdf.right)
ecdf.avg.right.exceed &lt;- colMeans(emp.cdf.right.exceed)
ecdf.avg.whole &lt;- c(ecdf.avg.left, ecdf.avg.center, ecdf.avg.right)
ecdf.avg.conf.int.left &lt;- apply(emp.cdf.left, 2, function(x) {t.test(x, conf.level = 0.999)$conf.int})
ecdf.avg.conf.int.right.exceed &lt;- apply(emp.cdf.right.exceed, 2, function(x) {t.test(x, conf.level = 0.999)$conf.int})</code></pre>
<pre class="r"><code>pdf(&quot;../figures/ecdf_keep_cor_avg.pdf&quot;, height = 5, width = 5)

par(mar = c(4.5, 4.5, 2.5, 1))

plot(c(qnorm(quantile.left), qnorm(quantile.center), -qnorm(quantile.right)), ecdf.avg.whole, type = &quot;l&quot;, xlab = &quot;z-score&quot;, ylab = &quot;CDF&quot;, cex.lab = 2, lwd = 2, main = expression(&#39;(a): Average Emp. CDF&#39;), cex.main = 2)

lines(c(qnorm(quantile.left), qnorm(quantile.center), -qnorm(quantile.right)),
      c(quantile.left, quantile.center, pnorm(-qnorm(quantile.right))), lty = 2, col = &quot;blue&quot;, lwd = 2)

legend(&quot;bottomright&quot;, lty = c(1, 2), col = c(1, &quot;blue&quot;), legend = c(expression(bar(&quot;F&quot;[p])), expression(&#39;N(0,1)&#39;)), lwd = 2, cex = 1.75, bty = &#39;n&#39;, x.intersp = 0.5)

rect(xleft = c(range(qnorm(quantile.left))[1], range(-qnorm(quantile.right))[1]),
     xright = c(range(qnorm(quantile.left))[2], range(-qnorm(quantile.right))[2]),
     ybottom = c(range(quantile.left, ecdf.avg.left)[1], range(pnorm(-qnorm(quantile.right)), ecdf.avg.right)[1]),
     ytop = c(range(quantile.left, ecdf.avg.left)[2], range(pnorm(-qnorm(quantile.right)), ecdf.avg.right)[2]),
     border = &quot;red&quot;, lty = c(1, 5)
     )</code></pre>
<pre class="r"><code>pdf(&quot;../figures/ecdf_keep_cor_avg_left.pdf&quot;, height = 5, width = 5)

par(mar = c(4.5, 4.5, 2.5, 1))

plot(qnorm(quantile.left), log(ecdf.avg.left), type = &quot;l&quot;,
     ylim = range(log(quantile.left), log(ecdf.avg.left)),
     xlab = &quot;z-score&quot;, ylab = &quot;log(CDF)&quot;, bty = &quot;n&quot;, lwd = 2, cex.lab = 2,
     main = expression(&#39;(b): Left Tail&#39;), cex.main = 2)

lines(qnorm(quantile.left), log(quantile.left), lty = 2, col = &quot;blue&quot;, lwd = 2)
lines(qnorm(quantile.left), log(pnorm(qnorm(quantile.left), 0, 1.1)), lty = 3, col = &quot;orange&quot;, lwd = 2)
lines(qnorm(quantile.left), log(pnorm(qnorm(quantile.left), 0, 1.05)), lty = 4, col = &quot;green&quot;, lwd = 2)

polygon(x = c(qnorm(quantile.left), rev(qnorm(quantile.left))),
        y = c(pmax(log(pmax(ecdf.avg.conf.int.left, 0)[1, ]), -30), rev(log(pmax(ecdf.avg.conf.int.left, 0)[2, ]))),
        border = NA,
        col = grDevices::adjustcolor(&quot;grey75&quot;, alpha.f = 0.5))

legend(&quot;bottomright&quot;, lty = c(1, 2, 4, 3), col = c(&quot;black&quot;, &quot;blue&quot;, &quot;green&quot;, &quot;orange&quot;), legend = c(
  expression(bar(&quot;F&quot;[p])),
  expression(N(0, 1)),
  expression(N(0, 1.05^2)),
  expression(N(0, 1.1^2))
), bty = &#39;n&#39;, lwd = 2, cex = 1.75, x.intersp = 0.5)

box(col = &quot;red&quot;)</code></pre>
<pre class="r"><code>pdf(&quot;../figures/ecdf_keep_cor_avg_right.pdf&quot;, height = 5, width = 5)

par(mar = c(4.5, 4.5, 2.5, 1))

plot(-qnorm(quantile.right), log(ecdf.avg.right.exceed), type = &quot;l&quot;,
     ylim = range(log(quantile.right), log(ecdf.avg.right.exceed)),
     xlab = &quot;z-score&quot;, ylab = &quot;log(1-CDF)&quot;, bty = &quot;n&quot;, lwd = 2, cex.lab = 2,
     main = expression(&#39;(c): Right Tail&#39;), cex.main = 2)

lines(-qnorm(quantile.right), log(quantile.right), lty = 2, col = &quot;blue&quot;, lwd = 2)
lines(-qnorm(quantile.right), log(pnorm(qnorm(quantile.right), 0, 1.1)), lty = 3, lwd = 2, col = &quot;orange&quot;)
lines(-qnorm(quantile.right), log(pnorm(qnorm(quantile.right), 0, 1.05)), lty = 4, lwd = 2, col = &quot;green&quot;)

polygon(x = c(-qnorm(quantile.right), rev(-qnorm(quantile.right))),
        y = c(pmax(log(pmax(ecdf.avg.conf.int.right.exceed[1, ], 0)), -30), rev(log(ecdf.avg.conf.int.right.exceed[2, ]))),
        border = NA,
        col = grDevices::adjustcolor(&quot;grey75&quot;, alpha.f = 0.5))

legend(&quot;bottomleft&quot;, lty = c(1, 2, 4, 3), col = c(&quot;black&quot;, &quot;blue&quot;, &quot;green&quot;, &quot;orange&quot;), legend = c(
  expression(bar(&quot;F&quot;[p])),
  expression(N(0, 1)),
  expression(N(0, 1.05^2)),
  expression(N(0, 1.1^2))
), bty = &#39;n&#39;, lwd = 2, cex = 1.75, x.intersp = 0.5)

box(col = &quot;red&quot;, lty = 5)</code></pre>
</div>
<div id="disclaimer" class="section level2">
<h2>Disclaimer</h2>
<p>The RNA-seq data set used in this file for realistic simulations was created from the real human liver tissue RNA-seq gene expression read counts. In particular, the sample and gene identifiers have been removed from the data.</p>
<p>The RNA-seq gene expression data were originally generated by the GTEx Project, which was supported by the Common Fund of the Office of the Director of the National Institutes of Health, and by NCI, NHGRI, NHLBI, NIDA, NIMH, and NINDS. The data used for the analyses described in this paper were obtained from the GTEx Portal at .</p>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.4.3 (2017-11-30)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS  10.14.1

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
 [1] workflowr_1.1.1   Rcpp_0.12.16      digest_0.6.15    
 [4] rprojroot_1.3-2   R.methodsS3_1.7.1 backports_1.1.2  
 [7] git2r_0.23.0      magrittr_1.5      evaluate_0.10.1  
[10] stringi_1.2.2     whisker_0.3-2     R.oo_1.22.0      
[13] R.utils_2.6.0     rmarkdown_1.9     tools_3.4.3      
[16] stringr_1.3.1     yaml_2.1.19       compiler_3.4.3   
[19] htmltools_0.3.6   knitr_1.20       </code></pre>
</div>

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
  This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
  analysis was created with
  <a href="https://github.com/jdblischak/workflowr">workflowr</a> 1.1.1
</p>
<hr>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
