---
title: "Large-scale realistic simulations with `cashr`: Stratified by noise"
author: "Lei Sun"
date: "2018-12-15"
output:
  workflowr::wflow_html:
    code_folding: show
---

- This file will take ~6 hours to knit.

```{r global_options, include = FALSE}
knitr::opts_chunk$set(warning = FALSE, message = FALSE)
```

```{r}
## load the data
r <- readRDS("../data/liver.rds")
```

```{r}
## load the script
source("../code/RNAseq_pipeline.R")
```

```{r}
## common simulation settings
ngene <- 1e4
nsamp <- 5
group <- rep(0 : 1, each = nsamp)
q <- 0.1 # nominal FDR
pi0 <- 0.9 # use this pi0 only
sim.theta <- function (ngene, pi0) {
  ## Gaussian only
  sample(c(
    rep(0, round(pi0 * ngene)),
    rnorm(round((1 - pi0) * ngene), 0, 2)
  ))
}
nsim <- 1e3

## choose the 1e4 most expressed genes
Y = lcpm(r)
subset = top_genes_index(ngene, Y)
r = r[subset,]

## multiple testing methods
method.name <- c("cashr", "BH", "qvalue", "ashr", "locfdr")
method.col <- scales::hue_pal()(length(method.names))[c(5, 1, 2, 4, 3)]
```

```{r}
XDP <- function (theta, qvalue, q) {
  disc <- (qvalue <= q)
  num.disc <- sum(disc)
  num.false.disc <- sum(theta[disc] == 0)
  num.true.disc <- sum(theta[disc] != 0)
  num.true <- sum(theta != 0)
  FDP <- num.false.disc / max(1, num.disc, na.rm = TRUE)
  TDP <- num.true.disc / num.true
  return(list(D = num.disc, FD = num.false.disc, FDP = FDP, TD = num.true.disc, TDP = TDP))
}
```

```{r cache = TRUE}
set.seed(777)

sim.FDP <- sim.TDP <- c()
Z.list <- disc.list <- list()

for (i in seq(nsim)) {
  ## simulate correlated null z-scores
  counts <- r[, sample(ncol(r), 2 * nsamp)]
  summary <- count_to_summary(counts, group)
  Z <- summary$z
  sd.Z <- sd(Z)
    
  ## simulate and normalize standard deviations
  s <- summary$s
  s <- s / sqrt(mean(s^2))
    
  ## simulate normal means theta
  theta <- sim.theta(ngene, pi0)
    
  ## construct observations
  X <- theta + s * Z
    
  ## test statistics
  z <- X / s
  p <- pnorm(-abs(z)) * 2
    
  ## apply different methods
  fit.BH <- p.adjust(p, method = "BH")
  XDP.BH <- XDP(theta, fit.BH, q)

  fit.qvalue <- qvalue::qvalue(p)
  qvalue.qvalue <- fit.qvalue$qvalues
  XDP.qvalue <- XDP(theta, qvalue.qvalue, q)
    
  fit.locfdr <- tryCatch(locfdr::locfdr(z, plot = 0), error = function(e) {NA})
  qvalue.locfdr <- tryCatch(ashr::qval.from.lfdr(fit.locfdr$fdr), error = function(e) {rep(NA, ngene)})
  XDP.locfdr <- XDP(theta, qvalue.locfdr, q)

  fit.ashr <- ashr::ash(X, s, mixcompdist = "normal", method = "fdr")
  qvalue.ashr <- ashr::get_qvalue(fit.ashr)
  XDP.ashr <- XDP(theta, qvalue.ashr, q)
    
  fit.cashr <- cashr::cash(X, s)
  qvalue.cashr <- fit.cashr$qvalue
  XDP.cashr <- XDP(theta, qvalue.cashr, q)
    
  ## FDP aggregated
  sim.FDP <- rbind.data.frame(sim.FDP, data.frame(
    sdZ = sd.Z,

    cashr = XDP.cashr$FDP,
    BH = XDP.BH$FDP,
    qvalue = XDP.qvalue$FDP,
    ashr = XDP.ashr$FDP,
    locfdr = XDP.locfdr$FDP
  ))
    
  ## TDP aggregated
  sim.TDP <- rbind.data.frame(sim.TDP, data.frame(
    sdZ = sd.Z,

    cashr = XDP.cashr$TDP,
    BH = XDP.BH$TDP,
    qvalue = XDP.qvalue$TDP,
    ashr = XDP.ashr$TDP,
    locfdr = XDP.locfdr$TDP
  ))
  
  ## noise stored
  Z.list[[i]] <- Z
  
  ## Discoveries stored
  disc.list[[i]] <- list(
    cashr = XDP.cashr,
    BH = XDP.BH,
    qvalue = XDP.qvalue,
    ashr = XDP.ashr,
    locfdr = XDP.locfdr
  )
}
```

```{r}
saveRDS(sim.FDP, '../output/sim.FDP.gaus.pi0.rds')
saveRDS(sim.TDP, '../output/sim.TDP.gaus.pi0.rds')
saveRDS(Z.list, '../output/sim.FDP.gaus.pi0.Z.rds')
saveRDS(disc.list, '../output/sim.FDP.gaus.pi0.disc.rds')
```

```{r}
sim.FDP <- readRDS("../output/sim.FDP.gaus.pi0.rds")
sim.TDP <- readRDS("../output/sim.TDP.gaus.pi0.rds")
Z.list <- readRDS("../output/sim.FDP.gaus.pi0.Z.rds")
disc.list <- readRDS("../output/sim.FDP.gaus.pi0.disc.rds")
```

```{r}
library(ggplot2)

boxplot.quantile <- function(x) {
  r <- quantile(x, probs = c(0.05, 0.25, 0.5, 0.75, 0.95))
  names(r) <- c("ymin", "lower", "middle", "upper", "ymax")
  return(r)
}

boxplot.quantile.sq <- function (x) {
  r <- sqrt(quantile(x^2, probs = c(0.05, 0.25, 0.5, 0.75, 0.95)))
  names(r) <- c("ymin", "lower", "middle", "upper", "ymax")
  return(r)
}

mean.sq <- function (x) {
  r <- sqrt(mean(x^2))
  return(r)
}

mysqrt_trans <- function() {
  scales::trans_new("mysqrt", 
                    transform = base::sqrt,
                    inverse = function(x) ifelse(x<0, 0, x^2),
                    domain = c(0, Inf))
}
```

## Distribution of FDP by distortion

```{r}
distortion <- cut(sim.FDP$sdZ, breaks = c(0, quantile(sim.FDP$sdZ, probs = 1 : 2 / 3), Inf), labels = c("Pseudo-deflation", "In-between", "Pseudo-inflation"))

sim.FDP.ext <- rbind.data.frame(
  cbind.data.frame(
    distortion,
    sim.FDP[, -1]
  ),
  cbind.data.frame(
    distortion = rep("All", length(distortion)),
    sim.FDP[, -1]
  )
)

FDP.ggdata <- reshape2::melt(sim.FDP.ext, id.vars = "distortion", variable.name = "method", value.name = "FDP")
```

```{r}
FDP.plot <- ggplot(data = FDP.ggdata, aes(x = method, y = FDP, fill = method, color = method)) +
  stat_summary(fun.data = boxplot.quantile.sq, geom = "boxplot", aes(width = 0.75), position = position_dodge(), show.legend = FALSE) +
  stat_summary(fun.y = mean.sq, geom = "point", position = position_dodge(width = 0.9), show.legend = FALSE, shape = 13, size = 3) +
  scale_x_discrete(limits = rev(levels(FDP.ggdata$method))) +
  scale_y_continuous(trans = "mysqrt", breaks = c(0, 0.1, 0.2, 0.4, 0.6, 0.8)) +
  coord_flip() +
  scale_color_manual(labels = method.name, values = method.col) +
  scale_fill_manual(labels = method.name, values = alpha(method.col, 0.35)) +
  facet_wrap(~distortion, nrow = 1) +
  geom_hline(yintercept = q, col = "black", linetype = "dashed", size = 1) +
  labs(x = expression(pi[0]), y = "FDP") +
  expand_limits(y = 0) +
  theme(plot.title = element_text(size = 12, hjust = 0),
        axis.title.y = element_blank(),
        axis.text.y = element_text(size = 15),
        axis.title.x = element_text(size = 15),
        axis.text.x = element_text(size = 10),
        strip.text = element_text(size = 15),
        legend.position = "bottom",
        legend.background = element_rect(color = "grey"),
        legend.text = element_text(size = 12)
        )
```

## Distribution of TDP by distortion

```{r}
sim.TDP.ext <- rbind.data.frame(
  cbind.data.frame(
    distortion,
    sim.TDP[, -1]
  ),
  cbind.data.frame(
    distortion = rep("All", length(distortion)),
    sim.TDP[, -1]
  )
)

TDP.ggdata <- reshape2::melt(sim.TDP.ext, id.vars = "distortion", variable.name = "method", value.name = "TDP")
```

```{r}
TDP.plot <- ggplot(data = TDP.ggdata, aes(x = method, y = TDP, fill = method, color = method)) +
  stat_summary(fun.data = boxplot.quantile, geom = "boxplot", aes(width = 0.75), position = position_dodge(), show.legend = FALSE) +
  coord_flip() +
  stat_summary(fun.y = mean, geom = "point", position = position_dodge(width = 0.9), show.legend = FALSE, shape = 13, size = 3) +
  scale_x_discrete(limits = rev(levels(TDP.ggdata$method))) +
  scale_color_manual(labels = method.name, values = method.col) +
  scale_fill_manual(labels = method.name, values = alpha(method.col, 0.35)) +
  facet_wrap(~distortion, nrow = 1) +
  labs(x = expression(pi[0]), y = "TDP") +
  expand_limits(y = 0) +
  theme(axis.title.y = element_blank(),
        axis.text.y = element_text(size = 15),
        axis.title.x = element_text(size = 15),
        axis.text.x = element_text(size = 10),
        strip.text = element_text(size = 15),
        legend.position = "bottom",
        legend.background = element_rect(color = "grey"),
        legend.text = element_text(size = 12)
        )
```

## Disclaimer

The RNA-seq data set used in this file for realistic simulations was created from the real human liver tissue RNA-seq gene expression read counts. In particular, the sample and gene identifiers have been removed from the data.

The RNA-seq gene expression data were originally generated by the GTEx Project, which was supported by the Common Fund of the Office of the Director of the National Institutes of Health, and by NCI, NHGRI, NHLBI, NIDA, NIMH, and NINDS. The data used for the analyses described in this paper were obtained from the GTEx Portal at [the GTEx Portal](https://www.gtexportal.org).
